src/io/flutter/FlutterUtils.java

```diff
@@ -58,4 +61,15 @@ public static VirtualFile getRealVirtualFile(@Nullable PsiFile psiFile) {
   public static void openFlutterSettings(@Nullable Project project) {
     ShowSettingsUtilImpl.showSettingsDialog(project, FlutterConstants.FLUTTER_SETTINGS_PAGE_ID, "");
   }
+
+  /**
+   * Checks whether a given string is a valid Dart identifier.
+   *
+   * See: https://www.dartlang.org/guides/language/spec
+   * @param id the string to check
+   * @return true if a valid identifer, false otherwise.
+   */
+  public static boolean isValidDartdentifier(@NotNull String id) {
```

Comment:
Do you want to check for Dart keywords?
Here is the list we use in the protobuf code generator:
https://github.com/dart-lang/dart-protoc-plugin/blob/master/lib/names.dart#L246

Or if we don't care whether it's a Dart keyword, perhaps this method should be renamed?